includes/Exception.php

```diff
@@ -374,26 +374,33 @@ class PermissionsError extends ErrorPageError {
 	public $permission, $errors;
 
 	function __construct( $permission, $errors = array() ) {
-		global $wgLang;
-
 		$this->permission = $permission;
 
 		if ( !count( $errors ) ) {
-			$groups = array_map(
-				array( 'User', 'makeGroupLinkWiki' ),
-				User::getGroupsWithPermission( $this->permission )
-			);
-
-			if ( $groups ) {
-				$errors[] = array( 'badaccess-groups', $wgLang->commaList( $groups ), count( $groups ) );
-			} else {
-				$errors[] = array( 'badaccess-group0' );
-			}
+			$errors[] = self::prepareBadAccessErrorArray( $this->permission );
 		}
 
 		$this->errors = $errors;
 	}
 
+	/**
+	 * @param string $permission Name of action
+	 * @return array error message key and message params (optionally)
+	 */
+	public static function prepareBadAccessErrorArray( $permission ) {
```

Comment:
Why has this been moved out of the constructor? Did I miss this being used somewhere else? Seems to still just be used in one place -- but then what is the purpose of this change?